editor/src/core/shared/uid-utils.ts

```diff
@@ -33,6 +33,15 @@ export function generateUID(existingIDs: Array<string>): string {
   return generateUID(existingIDs)
 }
 
+export const GeneratedUIDSeparator = `-`
+export function createIndexedUid(originalUid: string, index: string | number): string {
```

Comment:
to make things a little safer, I moved the creation of the dynamic uid to this dedicated function, and made an `extractOriginalUidFromIndexedUid` so if we change the format of the dynamic uid, we should be able to control this choke point here

Reply:
:+1: